blob/azureblob/azureblob.go

```diff
@@ -108,25 +111,6 @@ type Options struct {
 	SASToken SASToken
 }
 
-// Azure does not handle backslashes in the blob key well. As a workaround, all
```

Comment:
Note to reviewer: I removed this behavior because I don't think it's what we want. It basically made it so that you could create blob keys with "\\" in the name and they would be automagically munged to "/". But other providers don't do this, so it caused inconsistencies across providers.

Instead, I now escape "\\" if it's found in the raw blob key. Users should be using "/" as a delimiter with the `blob` package (and `fileblob` will convert "/" to `os.PathSeparator` if needed).